src/raftstore/store/peer.rs

```diff
@@ -353,6 +358,52 @@ impl Peer {
         Ok(())
     }
 
+    pub fn step(&mut self, m: eraftpb::Message) -> Result<()> {
+        if self.is_leader() && m.get_from() != INVALID_ID {
+            self.peer_heartbeats.insert(m.get_from(), Instant::now());
+        }
+        try!(self.raft_group.step(m));
+        Ok(())
+    }
+
+    pub fn check_peers(&mut self) {
+        if !self.is_leader() {
+            self.peer_heartbeats.clear();
+            return;
+        }
+
+        let peers = self.get_store().get_region().get_peers().to_owned();
```

Comment:
should we handle this every heartbeat? and I worry the performance if we have lots of leader regions too. 
I think we only need to handle peer_heartbeats for:
1. create the peer when store starts
2. apply the snapshot
3. do conf change